### PR TITLE
apscheduler 3.10.4

### PR DESCRIPTION
**Pull request by skupr-anaconda**
**Destination channel:** main

### Links

- [PKG-5118](https://anaconda.atlassian.net/browse/PKG-5118)
- release-notes: https://github.com/agronholm/apscheduler/releases
- release-diff: https://github.com/agronholm/apscheduler/compare/3.9.0...3.10.4
- license: https://github.com/agronholm/apscheduler/blob/master/LICENSE.txt
- requirements-tagged:
  - https://github.com/agronholm/apscheduler/blob/3.10.4/setup.py

### Explanation of changes:

- Clean up the recipe
- Add importlib-metadata to run and remove setuptools from runtime

[PKG-5118]: https://anaconda.atlassian.net/browse/PKG-5118?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by ifitchet — APPROVED** [review]
LGTM

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,33 +1,32 @@
...
-  script: {{ PYTHON }} -m pip install . -vv
+  script: {{ PYTHON }} -m pip install . --no-deps --no-build-isolation --ignore-installed --no-cache-dir -vv
 
 requirements:
   host:
     - python
     - pip
     - setuptools
-    - setuptools_scm
+    - setuptools-scm
```
Comment (lorepirri) [inline recipe/meta.yaml:22]: we can to import `setuptools_scm` in https://github.com/AnacondaRecipes/setuptools_scm-feedstock/pull/9/files

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,33 +1,32 @@
...
     - wheel
   run:
     - python
-    - setuptools >=0.7
     - six >=1.4.0
     - pytz
     - tzlocal >=2.0,!=3.*
+    - importlib-metadata >=3.6.0  # [py<38]
```
Comment (lorepirri) [inline recipe/meta.yaml:30]:
```suggestion
  run_constrained:
    - pymongo >= 3.0
    - redis >= 3.0
    - rethinkdb >= 2.4.0
    - sqlalchemy >= 1.4
    - tornado >= 4.3

```

**Review by lorepirri — COMMENTED** [review]
why the `entry_points` are not added in this case? Also Grayskull skips them

**Review by lorepirri — APPROVED** [review]

File: recipe/meta.yaml (hunk 1 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml:22]: Yes, but then both variants will be in the environment `setuptools_scm` and `setuptools-scm`. I think one variant `setuptools-scm` is enough for building the package

**Comment by skupr-anaconda** [comment]
> * https://github.com/agronholm/apscheduler/blob/3.10.4/setup.py

Because we put only `console_scripts` to `entry_points`

File: recipe/meta.yaml (hunk 1 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml:22]: @lorepirri Or should I follow the same practice we did?

Comment (skupr-anaconda) [reply recipe/meta.yaml:22]: Ah, I've checked the recipes and any variant can be used. No common practice

Comment (lorepirri) [reply recipe/meta.yaml:22]: all good, thanks for checking!

**Pull request merged**